tests/crypto.rs

```diff
@@ -198,79 +113,35 @@ fn keccak256() {
-        script.push(Opcode::MOVI(0x21, *b as Immediate18));
-        script.push(Opcode::SB(REG_HP, 0x21, (i + 1) as Immediate12));
-    });
-
-    // Set message address to 0x30
-    script.push(Opcode::ADDI(0x30, REG_HP, 1));
-
-    // Set hash address to 0x31
-    script.push(Opcode::ADDI(0x31, 0x30, length));
-
-    // Set computed hash address to 0x32
-    script.push(Opcode::ADDI(0x32, 0x31, 32));
-
-    // Set message length to 0x33
-    script.push(Opcode::MOVI(0x33, length as Immediate18));
-
-    // Set hash length to 0x34
-    script.push(Opcode::MOVI(0x34, 32));
-
-    // Compute the Keccak256
-    // m[computed hash] := keccack256(m[message, length])
-    // r[0x10] := m[hash] == m[computed hash]
-    script.push(Opcode::K256(0x32, 0x30, 0x33));
-    script.push(Opcode::MEQ(0x10, 0x31, 0x32, 0x34));
-    script.push(Opcode::LOG(0x10, 0, 0, 0));
-
-    // Corrupt the message
-    // m[message][0] := !m[message][0]
-    script.push(Opcode::LB(0x10, 0x30, 0));
-    script.push(Opcode::NOT(0x10, 0x10));
-    script.push(Opcode::SB(0x30, 0x10, 0));
-
-    // Compute the Keccak256
-    // m[computed hash] := keccack256(m[message, length])
-    // r[0x10] := m[hash] == m[computed hash]
-    script.push(Opcode::K256(0x32, 0x30, 0x33));
-    script.push(Opcode::MEQ(0x10, 0x31, 0x32, 0x34));
-    script.push(Opcode::LOG(0x10, 0, 0, 0));
-
-    script.push(Opcode::RET(REG_ONE));
-
-    let tx = Transaction::script(
-        gas_price,
-        gas_limit,
-        maturity,
-        script.into_iter().collect(),
-        vec![],
-        vec![],
-        vec![],
-        vec![],
-    )
-    .check(height, &params)
-    .expect("failed to check tx");
-
-    let receipts = Transactor::new(storage, Default::default())
-        .transact(tx)
-        .receipts()
-        .expect("Failed to execute script!")
-        .to_owned();
-
-    assert_eq!(receipts[0].ra().expect("$ra expected in receipt"), 1);
-    assert_eq!(receipts[1].ra().expect("$ra expected in receipt"), 0);
+    #[rustfmt::skip]
+    let script = vec![
+        Opcode::gtf(0x20, 0x00, GTFArgs::ScriptData),
+        Opcode::ADDI(0x21, 0x20, message.len() as Immediate12),
+        Opcode::MOVI(0x10, Bytes32::LEN as Immediate18),
+        Opcode::ALOC(0x10),
+        Opcode::ADDI(0x11, REG_HP, 1),
+        Opcode::MOVI(0x12, message.len() as Immediate18),
+        Opcode::K256(0x11, 0x20, 0x12),
+        Opcode::MEQ(0x13, 0x11, 0x21, 0x10),
+        Opcode::LOG(0x13, 0x00, 0x00, 0x00),
+        Opcode::RET(REG_ONE),
+    ].into_iter().collect();
+
+    let script_data = message.iter().copied().chain(hash.iter().copied()).collect();
+
+    let tx = TransactionBuilder::script(script, script_data)
+        .gas_price(gas_price)
+        .gas_limit(gas_limit)
+        .maturity(maturity)
+        .finalize_checked(height, &params);
+
+    let receipts = client.transact(tx);
+    let success = receipts.iter().any(|r| matches!(r, Receipt::Log{ ra, .. } if ra == &1));
```

Comment:
nit: Instead of comparing address with `ra`, can we compare values, e.g.:

```suggestion
    let success = receipts.iter().any(|r| matches!(r, Receipt::Log{ ra, .. } if *ra == 1));
```

Reply:
fixed all of these, agreed